.github/workflows/test-linux-smacv2.yml

```diff
@@ -0,0 +1,41 @@
+name: SMACv2 Tests on Linux
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+  workflow_dispatch:
+
+concurrency:
+  # Documentation suggests ${{ github.head_ref }}, but that's only available on pull_request/pull_request_target triggers, so using ${{ github.ref }}.
+  # On master, we want all builds to complete even if merging happens faster to make it easier to discover at which point something broke.
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/main' && format('ci-master-{0}', github.sha) || format('ci-{0}', github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  unittests:
+    if: ${{ github.event.label.name == 'Environments' }}
```

Comment:
That is above no? I didn’t modify anything here just added the line w if

Reply:
The thing i wonder is if this will be run on main

Reply:
I don't think it will
To me, the `on` block will go first and check that it's on main. Then the job `unittests` will be run but the `if` statement will always be false.
I think this condition goes under `on: pull_request`

Reply:
i don't think conditions can be put there